btree/btree.c

```diff
@@ -7846,10 +8605,14 @@ static int btree_kv_put_cb(struct m0_btree_cb *cb, struct m0_btree_rec *rec)
 
 	m0_bufvec_cursor_init(&scur, &datum->key->k_data);
 	m0_bufvec_cursor_init(&dcur, &rec->r_key.k_data);
+	rec->r_key.k_data.ov_vec.v_count[0] =
+	m0_vec_count(&datum->key->k_data.ov_vec);
```

Comment:
This should not be done by the consumer of btree code. The old code should not be changed. This initialization in the rec should have been done by the node_make() or node_rec().

Reply:
No change needed